registry/registry.go

```diff
@@ -24,6 +25,9 @@ var (
 	// ErrAlreadyExists is an error returned if an image being pushed
 	// already exists on the remote side
 	ErrAlreadyExists = errors.New("Image already exists")
+	// SessionID is a UUID that identifies the current execution context
+	// and is added to the docker headers in all requests against a registry
+	SessionID = uuid.Generate().String()
```

Comment:
That is really up for the context to decide. The client code should just pick it up from context and not really care.

I think session id may be shared across requests, or it wouldn't be that useful.